src/Pay/Adapter/Stripe.php

```diff
@@ -229,23 +229,71 @@ public function deleteCustomer(string $customerId): bool
         return $result['deleted'] ?? false;
     }
 
-    public function createFuturePayment(string $customerId, array $paymentMethodTypes = ['card'], ?string $paymentMethodConfiguration = null): array
+    public function createFuturePayment(string $customerId, ?string $paymentMethod = null, array $paymentMethodTypes = ['card'], array $paymentMethodOptions = [], ?string $paymentMethodConfiguration = null): array
     {
         $path = '/setup_intents';
         $requestBody = [
             'customer' => $customerId,
             'payment_method_types' => $paymentMethodTypes,
         ];
 
+        if ($paymentMethod != null) {
+            $requestBody['payment_method'] = $paymentMethod;
+        }
+
         if ($paymentMethodConfiguration != null) {
             $requestBody['payment_method_configuration'] = $paymentMethodConfiguration;
+            $requestBody['automatic_payment_methods'] = [
+                'enabled' => 'true',
+            ];
+            unset($requestBody['payment_method_types']);
+        }
+
+        if (! empty($paymentMethodOptions)) {
+            $requestBody['payment_method_options'] = $paymentMethodOptions;
         }
 
         $result = $this->execute(self::METHOD_POST, $path, $requestBody);
 
         return $result;
     }
 
+    public function listFuturePayments(?string $customerId = null, ?string $pyamentMethodId = null): array
+    {
+        $path = '/setup_intents';
```

Comment:
How come all the functions use the same endpoint?

Reply:
I don't think they do, create and list is same with different HTTP method.